ubuntuStation/views.py

```diff
@@ -60,38 +50,62 @@ def home():
 def tutorial():
     return render_template('tutorial.html')
 
-@app.route('/login', methods=["GET", "POST"])
-@limiter.limit("50 per hour")
+@app.route('/login')
 def login():
-    #print(request.method)
-    form = LoginForm(request.form)
-    #print("Next: ", request.args.get('next'))
-    if form.validate_on_submit(): #and request.method=="POST":
-        #print("Form is valid")
-        '''
-        if form.username.data=="Admin" and form.password.data=="pass":
-        '''
-        user = User.query.filter_by(username = form.username.data).first()
-        if user is not None and user.verify_password(form.password.data):
-            login_user(UserManager(form.username.data))
-            #login_user(UserManager(form.username.data))
-        next = request.args.get('next')
-        if not is_safe_url(next):
-            return abort(400)
+    if current_user.is_authenticated:
         return redirect(next or url_for('home'))
-    return render_template('login.html', form=form)
+    google = get_google_auth()
+    auth_url, state = google.authorization_url(
+        Auth.AUTH_URI, access_type='offline')
+    session['oauth_state'] = state
+    return render_template('login.html', auth_url=auth_url)
 
-def is_safe_url(target):
-    ref_url = urlparse.urlparse(request.host_url)
-    test_url = urlparse.urlparse(urlparse.urljoin(request.host_url, target))
-    return test_url.scheme in ('http', 'https') and ref_url.netloc == test_url.netloc
+@app.route('/gCallback')
+def callback():
+    # Redirect user to home page if already logged in.
+    #if current_user is not None and current_user.is_authenticated():
+        #return redirect(url_for('home'))
+    if 'error' in request.args:
+        if request.args.get('error') == 'access_denied':
+            return 'You denied access.'
+        return 'Error encountered.'
+    if 'code' not in request.args and 'state' not in request.args:
+        return redirect(url_for('login'))
+    else:
+        # Execution reaches here when user has
+        # successfully authenticated our app.
+        google = get_google_auth(state=session['oauth_state'])
+        try:
+            token = google.fetch_token(
+                Auth.TOKEN_URI,
+                client_secret=Auth.CLIENT_SECRET,
+                authorization_response=request.url)
+        except HTTPError:
+            return 'HTTPError occurred.'
+        google = get_google_auth(token=token)
+        resp = google.get(Auth.USER_INFO)
+        if resp.status_code == 200:
+            user_data = resp.json()
+            email = user_data['email']
+            user = User.query.filter_by(email=email).first()
+            if user is None:
+                #user = User()
+                #user.email = email
+                return redirect(url_for('login'))
+            user.name = user_data['name']
```

Comment:
@carlosparadis , we don't use user.name anywhere else. I will be removing this. For some reason, `name` was a valid key in the json response that we get, however it no longer is and so we get an error. Fixing it here in 
https://github.com/erdl/survey_admin/issues/64